x/pagination.go

```diff
@@ -0,0 +1,45 @@
+package x
+
+import (
+	"net/http"
+	"strconv"
+)
+
+const paginationMaxItems = 500
+
+// ParsePagination parses limit and page from *http.Request with given limits and defaults.
+func ParsePagination(r *http.Request) (page, limit int) {
+	if offsetParam := r.URL.Query().Get("page"); offsetParam == "" {
+		page = 0
+	} else {
+		if offset64, err := strconv.ParseInt(offsetParam, 10, 64); err != nil {
+			page = 0
+		} else {
+			page = int(offset64)
+		}
+	}
+
+	if limitParam := r.URL.Query().Get("limit"); limitParam == "" {
+		limit = paginationMaxItems
+	} else {
+		if limit64, err := strconv.ParseInt(limitParam, 10, 64); err != nil {
+			limit = paginationMaxItems
+		} else {
+			limit = int(limit64)
+		}
+	}
+
+	if limit > paginationMaxItems {
+		limit = paginationMaxItems
+	}
+
+	if limit < 0 {
+		limit = 0
```

Comment:
LIMIT 0 would return empty datasets. Is this desired?

Reply:
Right this should be `1`, nice catch!